src/main/java/mekanism/common/CommonPlayerTickHandler.java

```diff
@@ -382,4 +388,29 @@ private FallEnergyInfo getFallAbsorptionEnergyInfo(LivingEntity base) {
 
     private record FallEnergyInfo(@Nullable IEnergyContainer container, FloatSupplier damageRatio, FloatingLongSupplier energyCost) {
     }
+
+    @SubscribeEvent
+    public void getBreakSpeed(BreakSpeed event) {
+        Player player = event.getPlayer();
+        float speed = event.getOriginalSpeed();
+
+        // Blasting unit speed check
+        ItemStack stack = player.getMainHandItem();
+        if (!player.isShiftKeyDown() && stack.getItem() instanceof ItemMekaTool tool) {
+            Map<BlockPos, BlockState> blocks = tool.getBlastedBlocks(player.level, player, stack, event.getPos(), event.getState());
+            if (blocks != null) {
+                float targetHardness = event.getState().getDestroySpeed(player.level, event.getPos());
+                AtomicReference<Float> maxHardness = new AtomicReference<>(targetHardness);
+                blocks.forEach((pos, state) -> {
+                    float hardness = state.getDestroySpeed(player.level, pos);
+                    if (hardness > maxHardness.get()) {
+                        maxHardness.set(hardness);
+                    }
+                });
```

Comment:
Rather than having to use an atomic reference here you could just do something like:
```java
blocks.entrySet().stream()
              .map(entry -> entry.getValue().getDestroySpeed(player.level, entry.getKey()))
              .reduce(targetHardness, Float::max);
```